src/extensions/UserDefinedFormControllerExtension.php

```diff
@@ -97,5 +121,154 @@ protected function createPartialSubmission()
 
         // Refresh session on start
         $page->getRequest()->getSession()->set(PartialSubmissionController::SESSION_KEY, $submissionID);
+        $page->getRequest()->getSession()->set(PasswordForm::PASSWORD_SESSION_KEY, $submissionID);
+    }
+
+    /**
+     * Gets the PartialFormSubmission based on the current session
+     */
+    protected function getPartialFormSubmission()
+    {
+        $session = Controller::curr()->getRequest()->getSession();
+        $partialID = $session->get(PartialSubmissionController::SESSION_KEY);
+
+        return PartialFormSubmission::get()->byID($partialID);
+    }
+
+    /**
+     * Redirect user to form start if EnablePartialSubmissions is true
+     */
+    public function partialIndex(HTTPRequest $request = null)
+    {
+        if (!$this->owner->EnablePartialSubmissions) {
+            return $this->owner->index($request);
+        }
+
+        return $this->owner->redirect($this->owner->Link('start'));
+    }
+
+    /**
+     * Verify route, for password entry
+     */
+    public function verify()
+    {
+        return $this->owner->customise([
+            'Form' => $this->VerifyForm(),
+        ])->renderWith([UserDefinedFormController::class . '_start', Page::class]);
+    }
+
+    /**
+     * Generate the PasswordForm
+     * @return Form
+     */
+    public function VerifyForm()
+    {
+        $partial = $this->getPartialFormSubmission();
+        $controller = PartialUserFormVerifyController::create();
+        $controller->setPartialFormSubmission($partial);
+
+        return $controller->getForm()
+            ->setFormAction($this->owner->link('VerifyForm'));
+    }
+
+    /**
+     * Start route
+     */
+    public function start(HTTPRequest $request = null)
+    {
+        return $this->owner->customise([
+            'Form' => $this->StartForm(),
+        ]);
+    }
+
+    /**
+     * Generate the StartForm
+     * @return Form
+     */
+    public function StartForm()
+    {
+        $actions = FieldList::create(
+            FormAction::create('goToOverview')->setTitle('Start')
+        );
+
+        $form = Form::create(
+            $this->owner,
+            'StartForm',
+            FieldList::create(),
+            $actions
+        )->addExtraClass('userform');
+
+        return $form;
+    }
+
+    /**
+     * Redirect to the overview page, creating a new PartialFormSubmission if necessary
+     */
+    public function goToOverview($data, $form)
+    {
+        // If partial submission which matches this form already exists, redirect to overview
+        $submission = $this->getPartialFormSubmission();
+
+        if ($submission && $submission->ParentID === $this->owner->ID) {
+            return $this->owner->redirect($this->owner->Link('overview'));
+        }
+
+        // Else create new partial submission before redirecting
+        $this->createPartialSubmission();
```

Comment:
Also update the comment in PartialSubmissionController#L74 after removing `OnAfterInit`. Todo comment isn't necessary anymore since we're planning to save it differently (save button instead of prev/next and disable submit button until saving is done).